src/Window.hpp

```diff
@@ -141,26 +141,28 @@ struct SWindowSpecialRenderData {
 };
 
 struct SWindowAdditionalConfigData {
-    std::string                 animationStyle        = std::string("");
-    CWindowOverridableVar<int>  rounding              = -1; // -1 means no
-    CWindowOverridableVar<bool> forceNoBlur           = false;
-    CWindowOverridableVar<bool> forceOpaque           = false;
-    CWindowOverridableVar<bool> forceOpaqueOverridden = false; // if true, a rule will not change the forceOpaque state. This is for the force opaque dispatcher.
-    CWindowOverridableVar<bool> forceAllowsInput      = false;
-    CWindowOverridableVar<bool> forceNoAnims          = false;
-    CWindowOverridableVar<bool> forceNoBorder         = false;
-    CWindowOverridableVar<bool> forceNoShadow         = false;
-    CWindowOverridableVar<bool> forceNoDim            = false;
-    CWindowOverridableVar<bool> noFocus               = false;
-    CWindowOverridableVar<bool> windowDanceCompat     = false;
-    CWindowOverridableVar<bool> noMaxSize             = false;
-    CWindowOverridableVar<bool> dimAround             = false;
-    CWindowOverridableVar<bool> forceRGBX             = false;
-    CWindowOverridableVar<bool> keepAspectRatio       = false;
-    CWindowOverridableVar<int>  xray                  = -1; // -1 means unset, takes precedence over the renderdata one
-    CWindowOverridableVar<int>  borderSize            = -1; // -1 means unset, takes precedence over the renderdata one
-    CWindowOverridableVar<bool> forceTearing          = false;
-    CWindowOverridableVar<bool> nearestNeighbor       = false;
+    std::string                     animationStyle        = std::string("");
+    CWindowOverridableVar<int>      rounding              = -1; // -1 means no
+    CWindowOverridableVar<bool>     forceNoBlur           = false;
+    CWindowOverridableVar<bool>     forceOpaque           = false;
+    CWindowOverridableVar<bool>     forceOpaqueOverridden = false; // if true, a rule will not change the forceOpaque state. This is for the force opaque dispatcher.
+    CWindowOverridableVar<bool>     forceAllowsInput      = false;
+    CWindowOverridableVar<bool>     forceNoAnims          = false;
+    CWindowOverridableVar<bool>     forceNoBorder         = false;
+    CWindowOverridableVar<bool>     forceNoShadow         = false;
+    CWindowOverridableVar<bool>     forceNoDim            = false;
+    CWindowOverridableVar<bool>     noFocus               = false;
+    CWindowOverridableVar<bool>     windowDanceCompat     = false;
+    CWindowOverridableVar<bool>     noMaxSize             = false;
+    CWindowOverridableVar<Vector2D> maxSize               = Vector2D(99999, 99999);
```

Comment:
if you're making it a prop, add it as one. If you're not making this a prop, don't make it a CWindowOverridableVar. Furthermore, since this is optional, make it such. `std::optional` exists.

Reply:
If I make it a prop, how do I make it override the window rule? could I just use applyDynamicRule when setting the prop?

Reply:
what?

Reply:
If the window has a rule from the config the setprop command won't work.

Reply:
I'm regarded, I wasn't using hyprctl from the build dir